modules/network/src/main/java/org/apache/ignite/internal/network/serialization/marshal/Cycles.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.network.serialization.marshal;
+
+import org.apache.ignite.internal.network.serialization.ClassDescriptor;
+
+/**
+ * Logic related to cycles in object graphs.
+ */
+class Cycles {
```

Comment:
Seems like an overkill, a class (and also an object) for one method. Should it maybe be static?

Reply:
Well, the main class (the marshaller) is already bloated. I am trying to explore any possibility to cut it to pieces. :) Hence the class. Also, it has a distinctive responsibility, different from the marshaller responsibility: it's just about tracking cycleables. In the future, we might widen the notion of non-cycleable classes, so `Cycles` would contain more logic (and more code).

One more object on the heap is not a problem at all as it's one per marshaller which will not be created in huge numbers.

I generally dislike any statics as they tend to spread through the system, and when you have a lot of them, you have a problem.

But for now it makes sense to leave it as a static... I'll do it shortly.

Reply:
Removed the class (after consideration, it's not worth a separate class yet)